pkg/apis/internal.acorn.io/v1/appspec.go

```diff
@@ -749,7 +749,7 @@ type Secret struct {
 	Name        string            `json:"name,omitempty"`
 	Description string            `json:"description,omitempty"`
 	Type        string            `json:"type,omitempty"`
-	Params      GenericMap        `json:"params,omitempty"`
+	Params      *GenericMap       `json:"params,omitempty"`
```

Comment:
Since this is a pointer everywhere I feel like the chance of panic is extremely. Like I'm reading all the changes and trying to see if you first checked for nil.  What if you add a GetData() to `GenericMap` that will first check if the receiver is nil and then return nil.  So then existing code that did `foo.DeployArgs` and just be changed to `foo.DeployArgs.GetData()` and still be safe.

Reply:
I like this. Added.